docs/source/amp.rst

```diff
@@ -0,0 +1,48 @@
+.. role:: hidden
+    :class: hidden-section
+
+Automatic Mixed Precision package - torch.cuda.amp
+==================================================
+
+.. automodule:: torch.cuda.amp
+.. currentmodule:: torch.cuda.amp
+
+``torch.cuda.amp`` provides convenience methods for "mixed precision:"  using ``torch.float16`` for some operations
+and ``torch.float32`` for others.  TODO discuss how we want to summarize this...right balance of technical and quick.
+
+The two ingredients of the mixed precision recipe are
+
+* `Autocasting`_, to ensure each operation runs in its optimal precision, and
+* `Gradient Scaling`_, to mitigate FP16 gradient underflow during the backward pass.
+
+By default, you don't need to call ``.half()`` on your model(s) or data to use the routines below.
```

Comment:
Again, need to refocus on just the AmpScaler for the moment. Likely want to reduce this documentation to reflect that. Can extend when the guard is in.

Reply:
Do we even want to expose this documentation as part of the Sphinx tree until the guard is also merged?

Reply:
Let's get the PR into shape assuming we do.

Reply:
This sentence is misleading. Kill it and the next sentence/paragraph. When the with guard is in we can revisit them.